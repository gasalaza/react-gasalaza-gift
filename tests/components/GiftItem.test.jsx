import { render } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe('Pruebas en GiftItem', () => {
  
    const titulo = 'Un título';
    const url = 'https://localhost/algo.jpg';

    test('debe de hacer match con el snapshot', () => {
        
        
        const {container } = render(<GiftItem title={titulo} url={url}/>);

        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        const {src, alt} = render(<GiftItem title={titulo} url={url}/>).getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(titulo);

    });

    test('debe de mostrar el título en el componente', () => {
        
        const {getByText} = render(<GiftItem title={titulo} url={url}/>);

        expect(getByText(titulo)).toBeTruthy();

    });

});
