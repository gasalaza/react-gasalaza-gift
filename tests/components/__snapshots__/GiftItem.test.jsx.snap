// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un título"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un título
    </p>
  </div>
</div>
`;
